### ISSUE TITLE
Typescript errors with `quasar dev` and @quasar/app@1.7.4

### ISSUE THREAD

**Issue by GordonBlahut** [quasarframework/quasar#6912]
**Describe the bug**
The following typescript errors are displayed in the console when running `quasar dev` with `@quasar/app@1.7.4`. They are not present with `@quasar/app@1.7.3` and below:

```
ERROR in D:/WebClient/ClientApp/node_modules/@quasar/app/types/ssr.d.ts(8,18):
TS2339: Property 'url' does not exist on type 'Request<ParamsDictionary, any, any, any>'.
ERROR in D:/WebClient/ClientApp/node_modules/@types/express/index.d.ts(98,128):
TS2694: Namespace '"D:/WebClient/ClientApp/node_modules/@types/express-serve-static-core/index"' has no exported member 'Query'.
ERROR in D:/WebClient/ClientApp/node_modules/@types/express/index.d.ts(99,17):
TS2707: Generic type 'ErrorRequestHandler' requires between 0 and 3 type arguments.
ERROR in D:/WebClient/ClientApp/node_modules/@types/express/index.d.ts(108,116):
TS2694: Namespace '"D:/WebClient/ClientApp/node_modules/@types/express-serve-static-core/index"' has no exported member 'Query'.
ERROR in D:/WebClient/ClientApp/node_modules/@types/express/index.d.ts(108,131):
TS2707: Generic type 'Request<P, ResBody, ReqBody>' requires between 0 and 3 type arguments.
ERROR in D:/WebClient/ClientApp/node_modules/@types/express/index.d.ts(109,123):
TS2694: Namespace '"D:/WebClient/ClientApp/node_modules/@types/express-serve-static-core/index"' has no exported member 'Query'.
ERROR in D:/WebClient/ClientApp/node_modules/@types/express/index.d.ts(109,138):
TS2707: Generic type 'RequestHandler<P, ResBody, ReqBody>' requires between 0 and 3 type arguments.
```

**Codepen/jsFiddle/Codesandbox (required)**
N/A

**To Reproduce**
1. `quasar upgrade -i` to ensure `@quasar/app@1.7.4` is installed
2. `quasar dev`
3. See errors in console

**Expected behavior**
No typescript errors.

**Screenshots**

**Platform (please complete the following information):**
OS: Windows_NT(10.0.18363) - win32/x64
Node: 10.15.3
NPM: 6.13.4
Yarn: 1.21.1

**Comment by IlCallo** [comment]
TL;DR: delete your `yarn.lock` and run `yarn install` to regenerate.

Context: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43434#issuecomment-607181516
Solution: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/40138#issuecomment-589867440

Let us know if it solves the problem for you

**Comment by GordonBlahut** [comment]
Deleting yarn.lock broke a ton of other stuff. I'll look into if I can just delete the sections related to express and/or express-serve-static-core.

**Comment by pdanpdan** [comment]
Deleting yarn.lock should not generate problems except in very special cases. Check what you are doing there.

**Comment by GordonBlahut** [comment]
Isn't the purpose of yarn.lock to make dependency versions deterministic? Deleting it and running `yarn install` caused all sorts of unwanted package upgrades.

**Comment by IlCallo** [comment]
Regenerating a yarn.lock will cause your dependency graph to recompute to the latest available packages satisfying your dependencies constraints.

Normally, when updating your dependencies, only the direct ones and the ones which doesn't satisfy anymore your constraints are updated.

That's why regenerating yarn.lock could cause some problems.

Note that yarn.lock purpose is to force **someone else** installation to be the same as yours. Not to be deterministic between yarn.lock refreshes.

That said, removing only "express-serve-static-core" from yarn.lock and run "yarn install" should work too. But you're only hiding the dust under the carpet: those errors will surface at some point

**Comment by IlCallo** [comment]
@GordonBlahut any news on this?

**Comment by GordonBlahut** [comment]
> Note that yarn.lock purpose is to force someone else installation to be the same as yours. Not to be deterministic between yarn.lock refreshes.  
  
Well yes, but not only other people, but also to ensure consistency when regenerating `node_modules`, or if you have multiple devices, CI machine, etc so that unexpected updates to sub-dependencies that suddenly break your build or introduce security issues. I'd never expect it to be deterministic between yarn.lock refreshes, but for that reason I don't think you'd normally want to ever regenerate yarn.lock.

That said, I ended up pinning all my direct dependencies to specific versions (which they probably should have been all along) and deleting yarn.lock and letting all sub-dependencies resolve as they may.

I think this issue can be closed as I don't think there's anything really actionable.

**Comment by IlCallo** [comment]
Ok, gonna close it then. Thank you for the feedback anyway!

**Comment by hipertracker** [comment]
It does not work. I created a  new fresh Quasar project with TypeScript. When I run `quasar dev -m pwa --modern` I get a lot of TS errors from node_modules. This folder is not excluded.
```
My settings:

❯ quasar i

Operating System - Darwin(19.5.0) - darwin/x64
NodeJs - 12.18.0

Global packages
  NPM - 6.14.4
  yarn - 1.22.4
  @quasar/cli - 1.0.7
  cordova - Not installed

Important local packages
  quasar - 1.12.4 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app - 1.9.6 -- Quasar Framework local CLI
  @quasar/extras - 1.8.2 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 2.6.11 -- Reactive, component-oriented view layer for modern web interfaces.
  vue-router - 3.2.0 -- Official router for Vue.js 2
  vuex - 3.4.0 -- state management for Vue.js
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed
  @babel/core - 7.10.2 -- Babel compiler core.
  webpack - 4.43.0 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 3.11.0 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - 4.3.1 -- A plugin for your Webpack build process, helping you generate a manifest of local files that workbox-sw should precache.
  register-service-worker - 1.7.1 -- Script for registering service worker, with hooks
  typescript - 3.9.5 -- TypeScript is a language for application scale JavaScript development

Quasar App Extensions
  @quasar/quasar-app-extension-testing - 1.0.0 -- A Quasar App Extension for managing Test Harnesses
  @quasar/quasar-app-extension-testing-unit-jest - 1.1.0-beta.2 -- A Quasar App Extension for running Jest tests
```

**Comment by IlCallo** [comment]
@hipertracker 
Quasar typings are currently incompatible with TS 3.9.x, please avoid using custom TS version and rely on the one provided by `@quasar/app` (currently 3.8.3).

We'll update it as soon as we pin-point the problem with the new TS version

**Comment by hipertracker** [comment]
OK. I switched to default TS 3.8.3. Now I have other errors:

```
These dependencies were not found:

* core-js/modules/es6.array.iterator in ./node_modules/quasar/src/install.js, ./node_modules/quasar/src/lang.js and 20 others
* core-js/modules/es6.function.name in ./node_modules/quasar/src/install.js, ./node_modules/quasar/src/components/icon/QIcon.js and 1 other
* core-js/modules/es6.number.constructor in ./node_modules/quasar/src/components/btn/QBtn.js, ./node_modules/quasar/src/components/drawer/QDrawer.js and 8 others
* core-js/modules/es6.object.assign in ./node_modules/quasar/src/ssr-update.js, ./node_modules/quasar/src/components/drawer/QDrawer.js and 3 others
* core-js/modules/es6.object.freeze in ./node_modules/quasar/src/install.js
* core-js/modules/es6.object.keys in ./node_modules/quasar/src/install.js, ./node_modules/quasar/src/lang.js and 20 others
* core-js/modules/es6.object.to-string in ./node_modules/quasar/src/install.js, ./node_modules/quasar/src/lang.js and 21 others
* core-js/modules/es6.regexp.replace in ./.quasar/client-entry.js, ./node_modules/quasar/src/lang.js and 5 others
* core-js/modules/es6.regexp.split in ./node_modules/quasar/lang/en-us.js, ./node_modules/quasar/src/components/icon/QIcon.js and 2 others
* core-js/modules/es6.regexp.to-string in ./node_modules/quasar/src/utils/colors.js
* core-js/modules/es6.string.fixed in ./node_modules/quasar/src/components/drawer/QDrawer.js, ./node_modules/quasar/src/components/header/QHeader.js
* core-js/modules/es6.string.includes in ./node_modules/quasar/src/install.js, ./node_modules/quasar/src/ssr-update.js and 5 others
* core-js/modules/es6.string.starts-with in ./node_modules/quasar/src/components/icon/QIcon.js
* core-js/modules/es6.symbol in ./node_modules/quasar/src/lang.js, ./node_modules/quasar/src/icon-set.js and 15 others
* core-js/modules/es7.array.includes in ./node_modules/quasar/src/install.js, ./node_modules/quasar/src/ssr-update.js and 5 others
* core-js/modules/es7.object.get-own-property-descriptors in ./node_modules/quasar/src/lang.js, ./node_modules/quasar/src/icon-set.js and 15 others
* core-js/modules/web.dom.iterable in ./node_modules/quasar/src/install.js, ./node_modules/quasar/src/lang.js and 20 others

To install them, you can run: npm install --save core-js/modules/es6.array.iterator core-js/modules/es6.function.name core-js/modules/es6.number.constructor core-js/modules/es6.object.assign core-js/modules/es6.object.freeze core-js/modules/es6.object.keys core-js/modules/es6.object.to-string core-js/modules/es6.regexp.replace core-js/modules/es6.regexp.split core-js/modules/es6.regexp.to-string core-js/modules/es6.string.fixed core-js/modules/es6.string.includes core-js/modules/es6.string.starts-with core-js/modules/es6.symbol core-js/modules/es7.array.includes core-js/modules/es7.object.get-own-property-descriptors core-js/modules/web.dom.iterable
No type errors found
Version: typescript 3.8.3, eslint 6.8.0
```

**Comment by hipertracker** [comment]
Hmm, looks like I have to use `--modern` to fix those errors. It is surprising because it means my Quasar app will not work with a bit older browsers.

**Comment by hipertracker** [comment]
Problem solved. I had to remove 'apollo' package from my package.json. I can still use it as the global package for generating TypeScript types from my GraphQL server. It just cannot be installed together with the rest of my project packages.

**Comment by IlCallo** [comment]
This is due to the migration to core-js 3 which the ecosystem is going through right now.
Installing core-js 3 as a direct dependency of yours should work as a workaround
